android/engine/src/main/java/org/smartregister/fhircore/engine/ui/login/LoginActivity.kt

```diff
@@ -54,6 +65,16 @@ class LoginActivity :
       configureViews(configurationRegistry.retrieveConfiguration(AppConfigClassification.LOGIN))
     }
 
+    // Check if Pin enabled and stored then move to Pin login
+    val isPinEnabled = loginViewModel.loginViewConfiguration.value?.enablePin ?: false
+    val stayUserNamePasswordLogin =
+      loginViewModel.sharedPreferences.read(FORCE_LOGIN_VIA_USERNAME, "").equals("true", true)
+    val lastPinExist = !loginViewModel.sharedPreferences.read(PIN_KEY, "").isNullOrEmpty()
+    if (isPinEnabled && lastPinExist && !stayUserNamePasswordLogin) {
```

Comment:
If user force login once then is it mandatory to set the pin again?

Reply:
yes. 
as per story criteria, app need to setup PIN every time user login with username/password, if user forgot the pin, it can be reset with this flow.